src/core/webidl.js

```diff
@@ -358,6 +359,38 @@ function renderWebIDL(idlElement, index) {
   addIDLHeader(idlElement);
   return parse;
 }
+
+/**
+ * Add data-dfn-for to the closest section if not present already.
+ * @param {HTMLPreElement} idlElement
+ * @param {ReturnType<typeof webidl2.parse>} parse
+ */
+function addDataDfnFor(idlElement, parse) {
+  const closestSection = idlElement.closest("section");
+  if (closestSection.hasAttribute("data-dfn-for")) return;
+
+  const topLevelEntities = ["dictionary", "interface", "callback interface"];
+
+  const dfnFors = [];
+  for (const { tokens } of parse) {
+    if (topLevelEntities.includes(tokens.base.type)) {
+      const dfnFor = tokens.name.value;
+      dfnFors.push(dfnFor);
+    }
+  }
+  if (dfnFors.length === 1) {
+    closestSection.dataset.dfnFor = dfnFors[0];
+  } else if (!dfnFors.length) {
+    return;
+  } else {
+    closestSection.dataset.dfnFor = "";
+    const options = dfnFors.map(dfnFor => `"${dfnFor}"`).join(", ");
```

Comment:
Maybe we should not suggest these options? User might want to add something else?

Reply:
yeah, don't suggest these yet... remember that this currently works today:

```
<section data-dfn-for="Whatever">
   
    <section>
    <pre class="idl">
        interface Whatever{};
        interface SomethingElse{};
    <pre>
    </section>
</section>
```

Reply:
actually, the above is probably bad practice and doesn't matter...